### ISSUE TITLE
Sidebar niggles

### ISSUE THREAD

**Issue by edwh** [Freegle/iznik-nuxt#130]
- [ ] The jobs sidebar is slow to appear, because the data comes from a third party website.  Should we have a slide transition or something to make its appearance less disruptive?  Should we try to sync its appearance with the volunteer ops part of the sidebar?
- [ ] The sidebars are a bit brutally cut off at the bottom.  Make it look nicer.
-

**Comment by aaronferris** [comment]
This is due to the way the sidebar component has a fixed height, with the contained components also having a fixed height with an overflow. 

My recommendation would be to add a bottom border to each component within the sidebar.

Question though, are we sure we want to keep the border radius of the cards within the sidebar? Looks a bit messy to me.

![Screenshot 2020-01-02 at 10 18 59](https://user-images.githubusercontent.com/4068825/71662503-9e4e1000-2d49-11ea-87a6-6732f26cd107.png)

**Comment by edwh** [comment]
I agree the border radius is messy.

If we add a bottom border to each component in the sidebar won't we 
waste space within the sidebar?

------ Original Message ------

**Comment by aaronferris** [comment]
We will, 2px of space per overall component (IE 2 px for the volunteer component, 2px for 'x' component). I dont see any other way around it if we want to keep the approach we currently have (fixed height and overflow). The nature of this approach means the content that is too high for the parent element, will brutally be cut off.

**Comment by edwh** [comment]
I still don't quite get that - the height the children will vary, and 
you can scroll, so it's unlikely that the children will happen to fit 
neatly in the box showing the border.

What I used to have, but didn't like very much, was a kind of fade on 
the bottom of the box.  It looked ok on the left hand side where there 
was just one box, but looked weird where you had one on top of the 
other.

Or you can imagine some kind of subtle down arrow thing which is shown 
when you're not at the bottom.

Or we could decide to do nothing, which is always an option.

------ Original Message ------

**Comment by aaronferris** [comment]
I suppose we are talking at cross purposes, the issue states the content is being brutally cut off which in my mind was because it had a white bottom, then the wallpaper.

If the actual problem is that the content is half displayed on a particular card, then yes we will need to think of a way of displaying to the end user that its scrollable. From my POV, thats what the scrollbar is for.

**Comment by aaronferris** [comment]
Whatever we do a bottom border on the overall component definitely looks cleaner imo, either way.

**Comment by JayyajGH** [comment]
I've never been a fan of having multiple scrollbars on a page as it always tends to look a bit messy and I find the usability of it a bit weird.
To throw another option into the mix how about only displaying the two latest posts and then have a "Click to see more opportunities" anchor?

**Comment by JayyajGH** [comment]
I'd agree that that having the bottom border is definitely a step forward though - even if it doesn't resolve the full problem.

**Comment by edwh** [comment]
Yes, it's the half-displayed content I am concerned about.  Sorry for 
not being clear.

I agree the scrollbar indicates that, but that's quite subtle visually, 
and the part of the scrollbar which is more prominent and indicates the 
current position is up at the top.

So when I look at the screen, I see roughly the area below before it 
disappears into my peripheral vision.  The scrollbar is just grey which 
my eye ignores, and there's a very subtle down arrow.  The current 
position block for the top area is out of sight in peripheral vision, so 
my eye is drawn to the rather brutal way we cut girl guides in half.


Do you see what I mean?  A bottom border on the whole box might help, 
perhaps, both with this and the white/wallpaper aspect.

(The border-radius still needs to go.)

------ Original Message ------

**Comment by aaronferris** [comment]
Let's remove the border-radius, add the bottom border and see what you think from a branch. Leave the issue open for further discussion, then take it from there?

As @JayyajGH mentions the other option would be to display 'x' items then a 'read more' link which would resolve the issue but I would imagine a more significant piece of work.

**Comment by edwh** [comment]
I know what you mean about multiple scrollbars.  That adds to the 
feeling of clutter on this page on desktop which Chris has already 
highlighted, so I would like to keep kicking ideas around to see if we 
can make it feel simpler.

"Click to see more" that then renders them in that block or takes you to 
another page?  Jobs/Events/Volunteer Ops are a casual interest sort of 
thing, so I don't know (genuinely don't know) if people would be more 
likely to scroll rather than click away to another area.

I have this vague feeling that for mobiles and tablets swipey gestures 
are more natural than button clicks and page transitions.  Dunno if 
that's true/false.

------ Original Message ------

**Comment by edwh** [comment]
Aaron - yes, border-radius and bottom border will improve it.

------ Original Message ------

**Comment by aaronferris** [comment]
OK - ill take a look at that.

**Comment by JayyajGH** [comment]
My personal preference as a user would be for it to take you to a new page rather than try and display them all in the sidebar.

The only true test would be to find a CRO product that is free to charities/open source and do an A/B test.  I've used VWO and Optimizely before.  Probably a bit extreme in this situation at the moment though!

**Comment by aaronferris** [comment]
Ive gotten rid of the border-radius for the time being given

A) Seems to still be in discussion what how we want this to work
B) Adding a bottom border = ghost floating border until the component loads due to the way the markup is - would need further work in the component

**Comment by edwh** [comment]
I have some hand-rolled A/B test stuff (actually Bandit testing, see 
http://stevehanov.ca/blog/index.php?id=132) which I used for donation 
request testing.  I've not added anything into the new client for that 
yet, but will do in due course.

------ Original Message ------

**Comment by JayyajGH** [comment]
That's a really interesting read.  I'd not heard about that before.

**Comment by edwh** [comment]
Isn't it?  It's like A/B testing, but when you **_actually care_** about the result.
